Add first React components

luukveenis opened the pull request:
This adds the components necessary to render the checklists index page.

The main issue right now is that the table doesn't load until React updates for the first time (we wanted to have it at least render the table headings). This is because we need to specify the target container to render the React component into. If we specify the <table> tag as the element to render it into, it removes its children (the <thead> element and all its children) before inserting the <tbody> and containing elements. We should revisit this and see if there is a way to make this work.

luukveenis committed:
Remove unnecessary code

- config was not updated to reflect deleted files
  (readme.txt/BundleConfig.cs) resulting in build errors
- don't bundle react scripts since they need to be rendered on separate
  pages

luukveenis committed:
Add React components for checklist index page

luukveenis committed:
Expand tabs to spaces

Forgot to set global config for this, so JSX files were tabbed since there
is no syntax support for it.

luukveenis committed:
Add/fix basic styles

- .container bootstrap class was being overridden, removed this
- added classes for bootstrap tables to checklist table

sjlbos commented on LSCS.Web/Scripts/Templates/Checklists.jsx:
```diff
@@ -1 +1,44 @@
+                <td>{this.props.data.Title}</td>
+                <td>{this.props.data.Description}</td>
+                <td>{this.props.data.FileNumber}</td>
+                <td>{this.props.data.Location}</td>
```
Location is a composite object. Might need some wrangling to get a meaningful string value out of.

sjlbos commented on LSCS.Web/Scripts/Templates/Checklists.jsx:
```diff
@@ -1 +1,44 @@
+    render: function() {
+        return (
+            <tr>
+                <td>{this.props.data.Title}</td>
```
Do we need any of the other fields here? It's not the complete set.

luukveenis committed:
Added test class to generate JSON

** This is not a real test **

I needed a quick and dirty way to generate the JSON to POST to
the API to get some sample data into the DB for development. If you
want to use it, just run the test and it will output the JSON to stdout.

luukveenis committed:
Hook up server side data to React components

- Added "Access-Control-Allow-Origin" header to allow cross-origin
  resource sharing (else the web app can't hit the API)
- Loads in data asynchronously using XmlHTTPRequest
